src/invariants/token.ts

```diff
@@ -1,7 +1,47 @@
 import { BigNumber } from "bignumber.js";
 import { ERC20Contract } from "../wrappers";
 
+import * as singleLineString from "single-line-string";
+
+export const TokenAssertionErrors = {
+    MISSING_ERC20_METHOD: () =>
+        singleLineString`Contract does not implement ERC20 interface.`,
+};
+
 export class TokenAssertions {
+    // Throws an error if the given candidateContract does not implement the ERC20 interface.
+    public async implementsERC20(candidate: any): Promise<void> {
+        try {
+            await candidate.balanceOf.callAsync(candidate.address);
+            await candidate.totalSupply.callAsync();
+        } catch (error) {
+            throw new Error(TokenAssertionErrors.MISSING_ERC20_METHOD());
+        }
+
+        /*
+            Alternative, something like this could work"
+
+            // An ERC20 contract must at least implement the following methods:
+            const specMethods = [
+                'name',
+                'symbol',
+                'decimals',
+                'totalSupply',
+                'balanceOf',
+                'transfer',
+                'transferFrom',
+                'approve',
+                'allowance'
+            ];
+
+            specMethods.map((method: string) => {
+                if (!candidate.hasOwnProperty(method)) {
+                    throw new Error(TokenAssertionErrors.MISSING_ERC20_METHOD(method));
+                }
+            });
+        */
```

Comment:
Interesting -- I think doing the basic checks you've put in is sufficient -- let's add some sort of `NOTE` or `HACK` comment to explicate that we're only testing for a subset of the ERC20 methods.